### DISCUSSION TITLE
Issue with find and get not using the `_type` field

### DISCUSSION THREAD

**Discussion by calebwilson706** [sensedeep/dynamodb-onetable#421] [General]
**Describe the bug**

If you have two different models which use the same partitionKey and sortKey then the find and get commands will not respect the model type. For example:

If i had a SuccessfulFoo and FailedFoo models which both had the partition key `ID-${id}` and sort key `FOO_STATUS-${status}` along with other fields specific to the status. then one table's find command would still return some `FailedFoos` even if you did `successfulFoo.find`

I'm aware that the keys could be setup in a better way however I still believe that the find and get commands should automatically apply the same where clause as the scan which is `_type == EXPECTED_TYPE`

Sorry for the description, i'm finding it hard to explain but the last sentence should sum it up

**Comment by mobsense** [comment]
Please follow the issue template.

**Comment by mobsense** [comment]
Yes, that is a limitation. We expect that each model has a unique PK/SK combination. 

We could apply the _type as a filter, but that is an additional comparison for the vast majority of use cases. You could add that for your use case by a where clause.

**Comment by mobsense** [comment]
Thanks for the feedback. We really try to respond and fix issues quickly.

Doc is hard …

Can you point areas that need attention with the doc.

Can you help?

Regards

Michael



> I find it interesting that your documentation leaves out so many details, yet you have such a high standard for issues.
